operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -192,6 +198,45 @@ type ClusterSpec struct {
 	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
 }
 
+// ClusterConfiguration holds values (or references to values) that should be used
+// to configure the cluster. Where the cluster schema defines a non-string type for a
+// given key, the corresponding values here should be string-encoded (according to yaml
+// rules)
+// TODO: this type should be lifted to a shared module rather than duplicated in the `redpanda` CRD definition.
+type ClusterConfiguration struct {
+	Values map[string]ClusterConfigValue `json:"values,omitempty"`
+}
+
+type YamlRepresentation string
```

Comment:
```suggestion
type YAMLRepresentation string
```

nit but go's naming conventions would recommend this.

A comment and some examples would go a long way here too. Though it's possible example may cause more confusion as:
```
Foo
"Foo"
True
true
```

Are all valid

Reply:
I'll look to get at least one entry here with a kuttl test or something going, just to demonstrate that it works e2e.